### PR TITLE
Fix date display when comparing objects

### LINKED ISSUE #38: [Bug]: Date is displayed as 1 day before the actual date in the preview
### Expected behavior

Actual date should be displayed in the preview before merging.

### Actual behavior

Date is displayed as 1 day before the actual date in the preview before merging.

### Steps to reproduce

Select two objects which contain the date field for comparing
Click on the 'compare' button
In the preview, the date is displayed as 1 day before the actual date.

### PR DESCRIPTION
**Pull request by aryaantony92**
Resolves https://github.com/pimcore/object-merger/issues/38

### CODE REVIEW

File: src/Resources/public/js/panel.js [hunk 1]
```diff
@@ -257,15 +258,14 @@ pimcore.plugin.objectmerger.panel = Class.create({
...
             if (item) {
-                leftValue = item.value;
+                leftValue = (item.type && item.type == "date")?this.getFormattedDateFromTimestamp(item.data):item.value;
 
                 if (this.resultData[item.key]) {
                     rightValue = this.resultData[item.key].value;
                     changed = this.resultData[item.key].changed;
                 } else {
-                    rightValue = item.value2;
+                    rightValue = (item.type && item.type == "date")?this.getFormattedDateFromTimestamp(item.data):item.value2;
```
Comment (kingjia90) [inline src/Resources/public/js/panel.js:268]:
```suggestion
                    rightValue = (item.type && item.type == "date")?this.getFormattedDateFromTimestamp(item.data2):item.value2;
```
also it's covering only `date`, but doesn't work for `datetime` and any other custom types

**Comment by kingjia90** [comment]
Looks like the problem would be more broader than this, also version preview is affected
image
image
https://github.com/pimcore/pimcore/blob/4dc3b3e4280fb5cc92293224bff19c56ccd21444/models/DataObject/ClassDefinition/Data/Date.php#L169-L176
woud be fixed by using `$data->setTimezone('Europe/Stockholm');` before `$data->format('Y-m-d');`

In the example above, the timestamp is persisted as `1697493600`  which is 
GTM | Mon Oct 16 2023 22:00:00 GMT+0000
Time Zone | Tue Oct 17 2023 00:00:00 GMT+0200 (Central European Summer Time)

image

The solution might be around `getDiffVersionPreview` or `getVersionPreview` methods, otherwise custom object type (other than datetime) would be not covered by this fix

**Comment by dvesh3** [comment]
Might be fixed by https://github.com/pimcore/pimcore/issues/14658

**Comment by dvesh3** [comment]
Please see https://github.com/pimcore/object-merger/pull/42#issuecomment-1785074317

**Pull request closed without merging**